### ISSUE TITLE
[Bug] Error with attention mask, sliding window attention, and cached inference

### ISSUE THREAD

**Issue by mutiann** [fla-org/flash-linear-attention#431]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] My GPU is H100 and I have installed `triton-nightly` built by fla team, and double checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md)
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

An error willl occur when performing cached inference with attention mask on models with sliding window attention.  The sequence length of the cached KV is bounded by the window size, while the shape of `attention_mask` corresponds to the actual sequence length. The mismatch will trigger runtime error in the function `unpad_input` in `fla/layers/utils.py`.

A simple fix appears to be replacing 

https://github.com/fla-org/flash-linear-attention/blob/4c492817785d0fdf7c40d1848265164f1c47366b/fla/layers/utils.py#L142-L143

with 
```
batch_size, seq_len, *_ = states[0].shape
indices_k, cu_seqlens_k, max_seqlen_in_batch_k = get_unpad_data(attention_mask[:, -seq_len:])
```

### Steps to Reproduce the Bug

```
import fla
from transformers import AutoModelForCausalLM, AutoTokenizer, AutoConfig
import torch

name = 'fla-hub/transformer-340M-10B'


config = AutoConfig.from_pretrained(name)
config.window_size = 1024

tokenizer = AutoTokenizer.from_pretrained(name)
model = AutoModelForCausalLM.from_pretrained(
    name,
    config=config,
)
model = model.cuda()
model = model.bfloat16()

seq_lengths = [1025, 1026]
B = len(seq_lengths)
L = max(seq_lengths)
input_ids = torch.ones((B, L), dtype=torch.int64).cuda()
attention_mask = torch.zeros((B, L), dtype=torch.int64).cuda()
for i in range(B):
    attention_mask[i, -seq_lengths[i]:] = 1

r = model.generate(input_ids=input_ids, attention_mask=attention_mask, use_cache=True)
```

### Expected Behavior

Tokens should be generated without any error.

### Environment Information

1. Torch: 2.7.0
2. Triton: 3.3.0

**Comment by sustcsonglin** [comment]
Hi @mutiann  this issue has been fixed in https://github.com/fla-org/flash-linear-attention/commit/153e48a50f08d88081bd802c5d3ed9400671c57f
